src/util/psbt/map/global.rs

```diff
@@ -37,25 +37,10 @@ const PSBT_GLOBAL_VERSION: u8 = 0xFB;
 const PSBT_GLOBAL_PROPRIETARY: u8 = 0xFC;
 
 impl Map for PartiallySignedTransaction {
-    fn insert_pair(&mut self, pair: raw::Pair) -> Result<(), encode::Error> {
-        let raw::Pair {
-            key: raw_key,
-            value: raw_value,
-        } = pair;
-
-        match raw_key.type_value {
-            PSBT_GLOBAL_UNSIGNED_TX => return Err(Error::DuplicateKey(raw_key).into()),
-            PSBT_GLOBAL_PROPRIETARY => match self.proprietary.entry(raw::ProprietaryKey::from_key(raw_key.clone())?) {
-                btree_map::Entry::Vacant(empty_key) => {empty_key.insert(raw_value);},
-                btree_map::Entry::Occupied(_) => return Err(Error::DuplicateKey(raw_key).into()),
-            }
-            _ => match self.unknown.entry(raw_key) {
-                btree_map::Entry::Vacant(empty_key) => {empty_key.insert(raw_value);},
-                btree_map::Entry::Occupied(k) => return Err(Error::DuplicateKey(k.key().clone()).into()),
-            }
-        }
-
-        Ok(())
+    fn insert_pair(&mut self, _: raw::Pair) -> Result<(), encode::Error> {
+        // This method is only called during deserialization, for PSBT the insertion-like logic is
```

Comment:
I don't understand this comment, therefore I suspect it could be made more clear. Please remember that it's more necessary and helpful for people who do not understand what is going on, than to people that already do.

I understand that you figured out that this is never used, but I just don't understand *why* it is never used. And (often recrusive) "why" is the most important question to answer.

BTW. Implementation of a trait(interface) method that returns a "not implemented/needed" error is often a sign that the interface is incorrect. Possibly `trait Map` should be split into `trait  ROMap` and `trait Map : MutMap` or something.

Reply:
I suspect the reason is, as you say, that this `Map` trait tries to generalize over the different PSBT components but actually they have different logic.

Reply:
Thanks for your comments @dpc, I'll have a bit more of a play with this and see what I can come up with.